tests/ui/const_prop/issue-119908.rs

```diff
@@ -0,0 +1,7 @@
+// run-pass
+fn main() {
+    let x: [i32; 0] = [];
+    if x.len() > 0 {
+        x[0];
```

Comment:
does this trigger an unreachable/dead code code warning?

Reply:
Nope, lint is omitted for the whole block.